src/heitech.zer0mqXt.core/Patterns/Retry.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Threading.Tasks;
+using heitech.zer0mqXt.core.infrastructure;
+
+namespace heitech.zer0mqXt.core.patterns
+{
+    ///<summary>
+    /// Encapsulates the retry policy grasped from the configuration
+    ///</summary>
+    internal class Retry
+    {
+        private readonly SocketConfiguration _configuration;
+
+        internal Retry(SocketConfiguration configuration)
+        {
+            this._configuration = configuration;
+        }
+
+        ///<summary>
+        /// Register your retryable callback here
+        ///</summary>
+        internal async Task<XtResult<T>> RunAsyncWithRetry<T>(Func<Task<XtResult<T>>> retryableAction, string operationName = "request")
+            where T : class
+        {
+            try
+            {
+                return await retryableAction().ConfigureAwait(false);
+            }
+            catch (NetMQ.EndpointNotFoundException ntfnd)
+            {
+                _configuration.Logger.Log(new ErrorLogMsg($"NetMQ.Endpoint could not be found at {_configuration.Address()}: " + ntfnd.Message));
```

Comment:
What troubled me at first is that the `_configuration` is part of the `Retry`, but is not injected into the `retryableAction`. It is not ensured, that the `retryableAction` actually uses the same `_configuration` the `Address` of which is displayed in the error message.
But then I noticed that `Retry` it an internal type, so that the consumer does not need to be aware of this implicit relation.

Reply:
good catch, but yes that is intended :)